client/my-sites/sharing/connections/services/google-my-business.js

```diff
@@ -0,0 +1,156 @@
+					c-11.089-9.815-18.705-23.58-20.622-39.275l-2.135-17.483c0.322,2.624,0.476,5.21,0.476,7.77c0,35.235-28.61,64.72-64.836,64.72
+					c-35.801,0-64.81-29.022-64.81-64.81c0,35.789-29.022,64.81-64.823,64.81c-36.2,0-64.823-29.485-64.823-64.695
+					c0-2.586,0.167-5.223,0.476-7.86l-2.135,17.547c-1.917,15.694-9.532,29.459-20.634,39.275
+					c-11.102,9.828-25.664,15.733-41.487,15.733c-11.616,0-22.5-3.165-31.826-8.683v-30.012c9.327,5.519,20.21,8.683,31.826,8.683
+					c15.823,0,30.386-5.905,41.487-15.733c11.102-9.815,18.718-23.58,20.634-39.275l2.135-17.547c-0.309,2.637-0.476,5.274-0.476,7.86
+					c0,35.21,28.623,64.695,64.823,64.695c35.801,0,64.823-29.022,64.823-64.81c0,35.789,29.009,64.81,64.81,64.81
+					c36.226,0,64.836-29.485,64.836-64.72c0-2.56-0.154-5.146-0.476-7.77l2.135,17.483c1.917,15.694,9.532,29.459,20.622,39.275
+					c11.102,9.828,25.677,15.733,41.487,15.733C461.032,259.069,471.915,255.904,481.241,250.385z"
+				/>
+				<g>
+					<path
+						fill="#ACD1FC"
+						d="M385.171,188.508c0.322,2.624,0.476,5.21,0.476,7.77c0,35.235-28.61,64.72-64.836,64.72
+						c-35.801,0-64.81-29.022-64.81-64.81l-30.874-75.351L256,32.179h110.119l33.104,88.76l-14.065,67.401
+						C385.171,188.392,385.171,188.457,385.171,188.508z"
+					/>
+					<path
+						fill="#ACD1FC"
+						d="M145.869,32.181l28.658,97.416l-47.684,58.744v0.001l-2.148,17.649
+						c-1.917,15.694-9.532,29.459-20.634,39.275c-11.102,9.828-25.664,15.733-41.487,15.733c-11.616,0-22.5-3.165-31.826-8.683v-0.013
+						C12.337,241.419,0,221.363,0,198.426v-4.644L42.621,47.816c2.705-9.266,11.201-15.635,20.854-15.635H145.869z"
+					/>
+				</g>
+				<g>
+					<path
+						fill="#3A5BBC"
+						d="M481.241,252.315c-9.327,5.519-20.21,8.683-31.826,8.683c-15.81,0-30.385-5.905-41.487-15.733
+						c-11.089-9.815-18.705-23.58-20.622-39.275l-2.135-17.483c0-0.051,0-0.116-0.013-0.167l-19.039-156.16h82.395
+						c9.652,0,18.148,6.369,20.854,15.634L512,193.782v4.644C512,221.363,499.65,241.419,481.241,252.315z"
+					/>
+					<path
+						fill="#3A5BBC"
+						d="M256,32.181v164.007c0,35.789-29.022,64.81-64.823,64.81c-36.2,0-64.823-29.485-64.823-64.695
+						c0-2.586,0.167-5.223,0.476-7.86c0-0.039,0.013-0.064,0.013-0.103l19.026-156.16H256z"
+					/>
+				</g>
+				<path
+					fill="#FFFFFF"
+					d="M384.708,445.486c-35.481,0-64.347-28.866-64.347-64.347s28.866-64.347,64.347-64.347
+					c17.194,0,33.356,6.692,45.506,18.841l-13.644,13.644c-8.506-8.506-19.821-13.19-31.861-13.19
+					c-24.841,0-45.051,20.209-45.051,45.051c0,24.842,20.21,45.051,45.051,45.051c21.53,0,39.581-15.182,44.01-35.403h-44.011v-19.296
+					h64.347v9.648C449.055,416.619,420.189,445.486,384.708,445.486z"
+				/>
+			</svg>
+		);
+	}
+}
+
+export default connectFor(
+	GoogleMyBusiness,
+	( state, props ) => {
+		return {
+			...props,
+			removableConnections: props.keyringConnections,
+			fetchConnection: props.requestKeyringConnections,
+			siteUserConnections: props.keyringConnections.map( connection => ( {
```

Comment:
This creates a new array each time => props always changing

Reply:
I'm sorry, I thought it depends on `state`, but it doesn't seem to be the case, so why is it here?
In fact, none of the stuff here depends on `state`, maybe it's more appropriate for a c-tor?
I'm probably missing here something =)

Reply:
Did you notice the `connectFor`? It's not using `connect` here

Reply:
I noticed, yes, still think the point is valid...

Reply:
> I'm sorry, I thought it depends on state, but it doesn't seem to be the case, so why is it here?
It's just an helper to extend the `mapStateToProps` from `SharingService`.
So while they don't depend on `state`, they depend on the props computed in `connectFor`, which depends on the state. By doing so we override some props for our components. So everything related to publicize is thus ignored. Namely, making `removableConnections` and `siteUserConnections` limited to the keyring connections (it's normally publicize connection), replacing `fetchConnection` with `requestKeyringConnections` (instead of fetching publicize connections). Granted it's kind of a hack, but it's how it was implemented first and how Google Photos work.

Reply:
I see.
What about the always changing `siteUserConnections` prop?

Reply:
I pushed a commit: https://github.com/Automattic/wp-calypso/pull/23560/commits/e561b095447b889f895de71f5c563f085705b510